### ISSUE TITLE
[Lint/RedundantSafeNavigation] False positive when checking constant under a different module

### ISSUE THREAD

**Issue by ntkme** [rubocop/rubocop#13260]
It seems that rubocop assumes constant under a different module is always not `nil`, which is definitely not true:

``` ruby
module A
  X = nil

  X&.end_with?('test') # This has no warning
end

module B
  A::X&.end_with?('test') # This unexpectedly warns about Lint/RedundantSafeNavigation
end
```

```
W: [Correctable] Lint/RedundantSafeNavigation: Redundant safe navigation detected, use . instead.
  A::X&.end_with?('test') # This unexpected warns about Lint/RedundantSafeNavigation
      ^^
```
